var/spack/repos/builtin/packages/siesta/package.py

```diff
@@ -5,16 +5,38 @@
 
 
 import os
+import shutil
 
+from spack.build_systems import cmake
 from spack.package import *
 
 
-class Siesta(MakefilePackage):
+class Siesta(MakefilePackage, CMakePackage):
     """SIESTA performs electronic structure calculations and ab initio molecular
     dynamics simulations of molecules and solids.
     """
 
+    build_system(conditional("cmake", when="@5.0.0:"), "makefile", default="makefile")
+
     homepage = "https://departments.icmab.es/leem/siesta/"
+    git = "https://gitlab.com/siesta-project/siesta"
+
+    version(
+        "5.0.1",
+        url="https://gitlab.com/siesta-project/siesta/-/archive/5.0.1/siesta-5.0.1.tar.gz",
+        sha256="800a22a831c1d36c6f5fe4aa9c49ee510cbd49a0b2f87b3c8bf3edb6ebd0193a",
+    )
+
+    version(
+        "5.0.0",
+        url="https://gitlab.com/siesta-project/siesta/-/archive/rel-5.0/siesta-rel-5.0.tar.gz",
+        sha256="a407c31e8fd101efe4bfc68831870c524a8c397a39de7271d31ce0e6a4df92bb",
+    )
```

Comment:
The 5.x versions fail to build:
```suggestion
```

```py
var/spack/repos/builtin/packages/siesta/package.py:231, in build:
        230    def build(self, spec, prefix):
  >>    231        with working_dir("Obj"):
        232            make(parallel=False)
        233        if spec.satisfies("@:4.0.2"):
        234            with working_dir("Obj_trans"):
```
The setup phase of the 4.0.4 creates the `Obj` directory, but the setup for `5.x` does not.
Not with `build_system=cmake`, but also not with `build_system=makefile`.

I very heavily suggest removing the PR to an initial minimum PR and do not add the new versions and the `cmake` build system initially.

Reply:
The siesta 5.x versions use Cmake>3.20 unlike the older 4.x versions which use Make. Major changes have been made to the 5.x versions, these are no longer the same sources as the older 4.x versions. Now, the compilation is done in the same build directory as Siesta, and you no longer need the Obj directory.

Reply:
Thanks for the explanation. But this means that the way the build_system is selected to be either makefile or `cmake` is in needs of an update to ensure that `makefile` is not used for `@5:`:
```py
build_system(conditional("cmake", when="@5:"), conditional("makefile", when="@:4"))
```

Reply:
with the latest changes it's no longer necessary to specify the 'build_system'. It's automatically selects the correct argument for both versions 5.x and 4.x.

Reply:
Yes I see that, but

build_system(conditional("cmake", when="@5.0.0:"), "makefile", default="cmake")

still accepts `siesta@5 build_system=makefile` which then fails with `No such file or directory: 'Obj'`.

If you apply this change from https://github.com/CINES-HPC/spack/pull/1, spack would reject that and enforce that only build_system=cmake applies:
```py
build_system(conditional("cmake", when="@5:"), conditional("makefile", when="@:4"))
```
Also, this makes the intention and the facts that there is no choice and the fact that v4 only support `makefile` and v5 only supports `cmake` clear. Your current version lies about that.